ui/app/appointments/controllers/manage/appointmentsCreateController.js

```diff
@@ -14,7 +14,24 @@ angular.module('bahmni.appointments')
             $scope.timeRegex = Bahmni.Appointments.Constants.regexForTime;
             $scope.warning = {};
             $scope.minDuration = Bahmni.Appointments.Constants.minDurationForAppointment;
+            var ownAppointmentPrivilege = Bahmni.Appointments.Constants.privilegeOwnAppointments;
+            var manageAppointmentPrivilege = Bahmni.Appointments.Constants.privilegeManageAppointments;
+            var currentUserPrivileges = $rootScope.currentUser.privileges;
+
+            var getProviderForAppointmentPrivilegeUser = function (providers) {
+                if (appointmentCommonService.isCurrentUserHasPrivilege(manageAppointmentPrivilege, currentUserPrivileges)) {
+                    return providers;
+                }
+                if (appointmentCommonService.isCurrentUserHasPrivilege(ownAppointmentPrivilege, currentUserPrivileges)) {
+                    return _.filter(providers, function (provider) {
+                        return provider.uuid === $rootScope.currentProvider.uuid;
+                    });
+                }
+                return providers;
+            };
+
             $scope.appointmentCreateConfig = appointmentCreateConfig;
+            $scope.appointmentCreateConfig.providers = getProviderForAppointmentPrivilegeUser(appointmentCreateConfig.providers);
```

Comment:
This `$scope.appointmentCreateConfig.providers` seems not to used. And since this is the only place `getProviderForAppointmentPrivilegeUser(providers);` is used too, this questions it's purpose.

Reply:
Hi Larry,
This is basically to override the list of providers that show up while creating a new appointment.

Reply:
Thanks for the clarification